atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/SnapshotStoreImpl.java

```diff
@@ -49,16 +49,24 @@
     private final int maximumSize;
 
     @VisibleForTesting
-    SnapshotStoreImpl(int minimumSize, int maximumSize) {
+    SnapshotStoreImpl(int minimumSize, int maximumSize, CacheMetrics cacheMetrics) {
         this.snapshotMap = new TreeMap<>();
         this.timestampMap = new HashMap<>();
         this.liveSequences = MultimapBuilder.treeKeys().hashSetValues().build();
         this.minimumSize = minimumSize;
         this.maximumSize = maximumSize;
+        cacheMetrics.setSnapshotsHeldInMemory(snapshotMap::size);
+        cacheMetrics.setSequenceDifference(() -> {
+            if (snapshotMap.isEmpty()) {
+                return 0L;
+            } else {
+                return snapshotMap.lastKey().value() - snapshotMap.firstKey().value();
```

Comment:
Technically there is a race condition with the check above, but if this throws we'll survive